fence/resources/openid/ras_oauth2.py

```
@@ -46,6 +46,14 @@ def get_userinfo(self, token, userinfo_endpoint):
         access_token = token["access_token"]
         header = {"Authorization": "Bearer " + access_token}
         res = requests.get(userinfo_endpoint, headers=header)
+        if res.status_code != 200:
+            self.logger.info(
+                "Unable to get visa: status_code: {}, message: {}".format(
+                    res.status_code,
+                    str(res.json()),
```

i'd do something like `msg = res.text; try: msg = res.json(); except: pass` because you may not get json if there was an error